src/librustc_codegen_ssa/back/write.rs

```diff
@@ -145,9 +143,10 @@ impl ModuleConfig {
             || sess.opts.cg.linker_plugin_lto.enabled()
```

Comment:
Here's my understanding of the behaviour as the commit is written:
- rustc default: rlib contains `.o` with embedded bitcode section
- cargo default (`-Cbitcode-in-rlib=no`): rlib contains `.o` with no bitcode
- `-Clinker-plugin-lto`: rlib contains `.o`, which is bitcode
- `-Clto`: rlib contains `.o` with embedded bitcode section

It seems like the last one should be changed to be like `-Clinker-plugin-lto`?

Reply:
The first three I believe are correct, but the last isn't quite right. Currently the `-Clto` flag is basically ignored with rlibs AFAIK, and I believe that Cargo doesn't even pass `-Clto` to rlib compilations. Technically it's unfortunately a breaking change to alter the behavior of `-Clto` because, if accepted today, you can compile a `-Clto` rlib and then not do LTO in the final link step and it'll still work.

In any case though I'll be updating Cargo, after this lands, to more liberally use `-Clinker-plugin-lto` by default where possible when doing LTO builds.